src/python_testing/TC_CADMIN_1_3_4.py

```diff
@@ -0,0 +1,376 @@
+        attribute = GC_cluster.Attributes.BasicCommissioningInfo
+        duration = await self.read_single_attribute_check_success(endpoint=0, cluster=GC_cluster, attribute=attribute)
+        self.max_window_duration = duration.maxCumulativeFailsafeSeconds
+
+        self.step("3a")
+        # params = await self.openCommissioningWindow(dev_ctrl=self.th1, node_id=self.dut_node_id)
+        params = await self.th1.OpenCommissioningWindow(nodeid=self.dut_node_id, timeout=self.max_window_duration, iteration=1000, discriminator=1234, option=1)
+        self.print_step("TH1 params", params)
+
+        self.step("3b")
+        services = await self.get_txt_record()
+        if services.txt_record['CM'] != "2":
+            asserts.fail(f"Expected cm record value not found, instead value found was {str(services.txt_record['CM'])}")
+
+        self.step("3c")
+        BI_cluster = Clusters.BasicInformation
+        nl_attribute = BI_cluster.Attributes.NodeLabel
+        await self.write_nl_attr(th=self.th1, attr_val=nl_attribute)
+        await self.read_nl_attr(th=self.th1, attr_val=nl_attribute)
+
+        self.step(4)
+        # Establishing TH2
+        th2_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        th2_fabric_admin = th2_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.th1.fabricId + 1)
+        self.th2 = th2_fabric_admin.NewController(nodeId=2, useTestCommissioner=True)
+        await self.th2.CommissionOnNetwork(
+            nodeId=self.dut_node_id, setupPinCode=params.setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=1234)
+        rcac = self.th2.get_rcac()
+
+        th2_rcac_decoded = TLVReader(rcac).get()["Any"][9]
+
+        self.step(5)
+        # TH_CR1 reads the Fabrics attribute from the Node Operational Credentials cluster using a fabric-filtered read
+        th1_fabric_info = await self.get_fabrics(th=self.th1)
+
+        # Verify that the RootPublicKey matches the root public key for TH_CR1 and the NodeID matches the node ID used when TH_CR1 commissioned the device.
+        await self.send_single_cmd(dev_ctrl=self.th1, node_id=self.dut_node_id, cmd=Clusters.GeneralCommissioning.Commands.ArmFailSafe(10))
+        th1_cam_rcac = TLVReader(base64.b64decode(
+            self.certificate_authority_manager.activeCaList[0]._persistentStorage._jsonData["sdk-config"]["f/1/r"])).get()["Any"][9]
+        if th1_fabric_info[0].rootPublicKey != th1_cam_rcac:
+            asserts.fail("public keys from fabric and certs for TH1 are not the same")
+        if th1_fabric_info[0].nodeID != self.dut_node_id:
+            asserts.fail("DUT node ID from fabric does not equal DUT node ID for TH1 during commissioning")
+
+        # Expiring the failsafe timer in an attempt to clean up before TH2 attempt.
+        await self.th1.SendCommand(self.dut_node_id, 0, Clusters.GeneralCommissioning.Commands.ArmFailSafe(0))
```

Comment:
same Q about disarm - nothing in here should be communicating with the DUT, so what is the failsafe doing here?

Reply:
Thank you, you are correct, the fail safes were no longer needed with the new method of getting the RCAC back. 
The initial way had implemented caused need to implement the fail safes in these areas, but fail safes are no longer needed.